scipy/stats/tests/test_distributions.py

```diff
@@ -4315,6 +4318,99 @@ def test_burr_nan_mean_var_9544(self):
         assert_(np.isfinite(e4))
 
 
+class TestStudentizedRange:
+    # Critical Q values were selected from this table:
+    # https://link.springer.com/chapter/10.1007/978-1-4613-9629-1_12
+    # For alpha=.05 and .01, and for each value of v=[1, 3, 10, 20],
+    # a Q was picked from each table for k=[2, 8, 14, 20].
+    ks = [2, 8, 14, 20]
+    vs = [1, 3, 10, 20]
+    alpha05 = [17.97, 45.40, 54.33, 59.56,
+               4.501, 8.853, 10.35, 11.24,
+               3.151, 5.305, 6.028, 6.467,
+               2.950, 4.768, 5.357, 5.714]
+    alpha01 = [90.03, 227.2, 271.8, 298.0,
+               8.261, 15.64, 18.22, 19.77,
+               4.482, 6.875, 7.712, 8.226,
+               4.024, 5.839, 6.450, 6.823]
+    alphas = alpha05 + alpha01
+    ks_d_16 = np.ravel(list((repeat(ks, 4))))
+    vs_d_16 = np.ravel(list(zip(*(repeat(vs, 4)))))
+    ks_d = np.tile(ks_d_16, 2)
+    vs_d = np.tile(vs_d_16, 2)
+    ps = [.95] * 16 + [.99] * 16
+
+    path_prefix = os.path.dirname(__file__)
+    relative_path = "data/studentized_range_mpmath_ref.json"
+    with open(os.path.join(path_prefix, relative_path), "r") as file:
+        pregenerated_data = json.load(file)
+
+    @pytest.mark.parametrize("case_result", pregenerated_data["cdf_data"])
+    def test_cdf_against_mp(self, case_result):
+        print(case_result)
+        src_case = case_result["src_case"]
+        mp_result = case_result["mp_result"]
+        print(src_case)
+
+        res = stats.studentized_range.cdf(src_case["q"],
+                                          src_case["k"],
+                                          src_case["v"])
+
+        assert_allclose(res, mp_result,
+                        atol=src_case["expected_atol"],
+                        rtol=src_case["expected_rtol"])
+
+    def test_cdf_against_tables(self):
+        res_p = stats.studentized_range.cdf(self.alphas, self.ks_d, self.vs_d)
+        assert_array_almost_equal(res_p, self.ps, decimal=4)
+
+    @pytest.mark.slow
+    def test_ppf_against_tables(self):
+        res_q = stats.studentized_range.ppf(self.ps, self.ks_d, self.vs_d)
+        # some table values only provide up to the tenths place
+        assert_array_almost_equal(res_q, self.alphas, decimal=1)
```

Comment:
That's why this should be a relative tolerance. There are four decimal places given in the answer, so try rtol = 1e-4.
Please change all `assert_array_almost_equal` to `assert_allclose` with appropriate relative tolerance (unless the values being compared are nearly zero, in which case you'll need an absolute tolerance).
`assert_allclose` is recommended by NumPy for new code. (See documentation for `assert_array_almost_equal`.)

Reply:
resolved